### PR TITLE
Add offline mode

### PR DESCRIPTION
**Pull request by stephaniehobson**
- worker installs at global scope
- saves js, css, and 4 key pages (including schedule)
- version control is done by ISO date of build
- for HTML it tries network first and falls back to cache
- for non-HTML it tries cache first
- image requests fall back to either: generic headshot, dataURI for hero image, or generic 'offline' image

### CODE REVIEW

File: source/worker.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+                    '/berlin-2016/schedule/',
+                    '/berlin-2016/code-of-conduct/',
+                    '/berlin-2016/'
+                ]);
+            }).catch(function() {
+                // problem with loading something, now what?
+            });
+    }
+
+        self.addEventListener('install', function (event) {
```
Comment (hoosteeno) [inline source/worker.js]: Nit: indentation.

File: README.md [hunk 2]
```diff
@@ -74,6 +74,10 @@ The site is deployed to staging and production using Docker and Deis, and you ca
   * useful for testing nginx.conf changes
 * `make serve`: run the nginx deployment image
   * listens on port 8000 by doefault, override with the SERVE_PORT environment variable
+* `make generate-cert`: create a self signed cert for localhost
+* `make serve https`: run the nginx deployment image but force https
```
Comment (hoosteeno) [inline README.md]: For me what works is `make serve-https`

**Comment by stephaniehobson** [comment]
Needs to take into account requests to "/" and forward to "/berlin-2016/"

**Comment by hoosteeno** [comment]
This PR is mostly working as advertised, notwithstanding the two nits in code comments. There is one bug that we may want to examine and workaround.

Steps to replicate:
* Load the site in a recent version of Firefox
* Stop the webserver
* Change some copy, build the site, start the webserver
* Load the site in Firefox again -- see, the copy changed!
* Stop the webserver
* Load the site in Firefox again -- whoa, what happened to the images?

Expected: 
* Images are cached

Actual:
* In Firefox (but not in Chrome), images are not cached after a change. Other assets may also be missing.

Notes:
* The storage inspector in Firefox shows all the prior caches, while the storage inspector in Chrome only shows the most recent cache. What are these browsers doing differently, and how can we work around it?

**Comment by stephaniehobson** [comment]
After no small amount of testing @hoosteeno and I are thinking there is maybe a FireFox bug/feature where the implicitly cached files are dropped when a new service worker is installed.

He's working on a test case to potentially open a bug.

In the mean time, the current experience is an improvement and we'd like to get it onto staging for testing and potential deployment.

**Pull request merged**